microsetta_admin/server.py

```diff
@@ -556,6 +623,125 @@ def new_kits():
                          mimetype='text/csv')
 
 
+def _read_csv_file(file):
+    content = file.read().decode('utf-8-sig')
+    return [row[0] for row in csv.reader(io.StringIO(content),
+                                         skipinitialspace=True)
+            if row]
+
+
+def _handle_add_barcodes_api_request(payload):
+    status, result = APIRequest.post('/api/admin/add_barcodes',
+                                     json=payload)
+    if status != 200 and status != 204:
+        return render_template('add_barcode_to_kit.html',
+                               error_message=result,
+                               **build_login_variables()), status
+    return result
+
+
+def _save_and_send_csv(kit_ids, barcodes):
+    with tempfile.NamedTemporaryFile(mode='w',
+                                     delete=False,
+                                     newline='') as file:
+        writer = csv.writer(file)
+        writer.writerow(['Kit IDs', 'Barcode'])
+        writer.writerows(zip(kit_ids, barcodes))
+        filename = file.name
+
+    stamp = datetime.now().strftime('%d%b%Y-%H%M')
+    fname = f'kits-{stamp}.csv'
+    return send_file(filename,
+                     as_attachment=True,
+                     download_name=fname)
+
+
+@app.route('/add_barcode_to_kit', methods=['GET', 'POST'])
+def new_barcode_kit():
+    if request.method == 'GET':
+        return render_template('add_barcode_to_kit.html',
+                               **build_login_variables())
+
+    kit_ids = []
+    barcodes = []
+
+    if 'kit_ids' in request.form:
+        kit_ids = request.form['kit_ids']
+
+    if 'user_barcode' in request.form:
+        user_barcode = request.form['user_barcode']
+        barcodes.append(user_barcode)
+    else:
+        user_barcode = None
+
+    generate_barcode_single = \
+        request.form.get('generate_barcode_single')
+    generate_barcodes_multiple = \
+        request.form.get('generate_barcodes_multiple')
+
+    if 'kit_ids' in request.files:
+        kit_ids_file = request.files['kit_ids']
+        kit_ids = _read_csv_file(kit_ids_file)
+
+    if 'barcodes_file' in request.files:
+        barcodes_file = request.files['barcodes_file']
+        barcodes = _read_csv_file(barcodes_file)
+
+    if generate_barcode_single or user_barcode:
```

Comment:
I feels like there's a lot of redundancy here. For example, we know that if the user selects Add Barcode to Single Kit, kit_ids should be present as a string and either user_barcode or generate_barcode_single should be present. Is it necessary to check for each of those individually?